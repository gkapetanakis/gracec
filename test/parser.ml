let () =
  let dirname = "programs/" in
  let filenames = [ "program1.grc" ] in
  let test filename =
    let chan = open_in (dirname ^ filename) in
    let lexbuf = Lexing.from_channel chan in
    try
      let ast = Gracec_lib.Parser.program Gracec_lib.Lexer.token lexbuf in
      Gracec_lib.Debug_utils.print_program 2 0 ast
    with
    | Gracec_lib.Lexer.Lexing_error msg -> prerr_endline msg
    | Gracec_lib.Parser.Error ->
        prerr_endline
          (Printf.sprintf "Parse error at line %d:\n" lexbuf.lex_curr_p.pos_lnum)
  in
  List.iter test filenames
