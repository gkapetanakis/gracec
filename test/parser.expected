MainFunc:
  Header:
    Name: main
    Params: none
    RetType: nothing
  LocalDefs:
    VarDef:
      Name: a
      Type: int
    VarDef:
      Name: b
      Type: char
    VarDef:
      Name: c
      Type:
        Array:
          Size: 5
          Of: int
    VarDef:
      Name: d
      Type:
        Array:
          Size: 10
          Of: char
  Block: empty
